src/main/resources/benchmarks/tpch/dialect-postgres.xml

```diff
@@ -3,65 +3,76 @@
     <dialect type="POSTGRES">
         <procedure name="Q1">
             <statement name="query_stmt">
+                /*monitor-Q01*/
```

Comment:
I wonder if these tweaks could be generalized so that they could automatically be added during dialect construction somehow.
It'd make it applicable beyond this particular workload as well.
For instance, what if each statement had `with_monitoring_comment="true"` property that defaulted to `"true"` if omitted.
Similarly, in the `DatabaseType` enum, each db type could have a `generate_monitoring_comments=true` default, that could be overridden to `false` if that particular DB doesn't accept `/* monitoring-NAME */` style syntax.
Finally, in `Procedure.java` modify `getStatements()` and `setSql()` in `SQLStmt.java` somehow to adjust the query dynamically to include that `/* monitoring-{name} */` comment in the query text?

I feel like this might be more maintainable and generic for the long run.  Thoughts?

Reply:
BTW, this is a guess on how that implementation might work.  There are various layers of indirection here I haven't totally followed thru yet.

Reply:
And such a system would also need some tests.

Reply:
Hmm, good point, I think this could work. In essence, we need the monitoring comment to filter the cached query plans to only store those that are actually part of the benchmark. I'll play around with it to see what I can come up with.

Reply:
It took me a bit to figure out the best (non-intrusive and generally applicable) way to do this but I think I got it! Basically, I'm setting a flag in the workload configuration that triggers a statement update similar to the dialect adjustments that we currently have. That way, it's generic and applied for a) any enabled DBMS and b) for any benchmark.